browser/ui/views/wayback_machine_bubble_view.cc

```diff
@@ -0,0 +1,157 @@
+/* Copyright (c) 2024 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/ui/views/wayback_machine_bubble_view.h"
+
+#include <memory>
+#include <string>
+#include <utility>
+
+#include "base/functional/bind.h"
+#include "brave/browser/ui/views/page_action/wayback_machine_action_icon_view.h"
+#include "brave/components/brave_wayback_machine/brave_wayback_machine_tab_helper.h"
+#include "brave/components/l10n/common/localization_util.h"
+#include "brave/grit/brave_generated_resources.h"
+#include "chrome/browser/ui/browser.h"
+#include "chrome/browser/ui/tabs/tab_strip_model.h"
+#include "chrome/browser/ui/views/chrome_layout_provider.h"
+#include "content/public/browser/render_frame_host.h"
+#include "ui/base/metadata/metadata_impl_macros.h"
+#include "ui/views/controls/label.h"
+#include "ui/views/layout/box_layout.h"
+
+namespace {
+
+BraveWaybackMachineTabHelper* GetTabHelper(content::WebContents* web_contents) {
+  if (!web_contents) {
+    return nullptr;
+  }
+
+  return BraveWaybackMachineTabHelper::FromWebContents(web_contents);
+}
+
+gfx::FontList GetFont(int font_size, gfx::Font::Weight weight) {
+  gfx::FontList font_list;
+  return font_list.DeriveWithSizeDelta(font_size - font_list.GetFontSize())
+      .DeriveWithWeight(weight);
+}
+
+}  // namespace
+
+// static
+void WaybackMachineBubbleView::Show(Browser* browser, views::View* anchor) {
+  auto* web_contents = browser->tab_strip_model()->GetActiveWebContents();
+  if (!web_contents) {
+    return;
+  }
+
+  auto* tab_helper = GetTabHelper(web_contents);
+  if (!tab_helper) {
+    return;
+  }
+
+  // Don't need to launch again if existed.
+  if (tab_helper->active_window()) {
+    return;
+  }
+
+  content::WebContents::Getter wc_getter(base::BindRepeating(
+      [](content::GlobalRenderFrameHostId rfh_id) {
+        return content::WebContents::FromRenderFrameHost(
+            content::RenderFrameHost::FromID(rfh_id));
+      },
+      web_contents->GetPrimaryMainFrame()->GetGlobalId()));
+  views::Widget* const widget = views::BubbleDialogDelegateView::CreateBubble(
+      std::make_unique<WaybackMachineBubbleView>(std::move(wc_getter), anchor));
```

Comment:
I think we can replace wc_getter with web_contents->GetWeakPtr(). wc_getter is a kind of legacy I believe.

```
  // Most uses of Getter and OnceGetter can likely be safety replaced with
  // base::WeakPtr<WebContents>.
```

Reply:
Oh, it's deprecated one. I didn't know. thanks!